### ISSUE TITLE
Slack post failure

### ISSUE THREAD

**Issue by xUrko** [jenkinsci/slack-plugin#479]
- Jenkins 2.160
- Slack plugin 2.14

java.lang.NullPointerException
	at java.util.regex.Matcher.getTextLength(Unknown Source)
	at java.util.regex.Matcher.reset(Unknown Source)
	at java.util.regex.Matcher.<init>(Unknown Source)
	at java.util.regex.Pattern.matcher(Unknown Source)
	at jenkins.plugins.slack.ActiveNotifier$MessageBuilder.extractReplaceLinks(ActiveNotifier.java:508)
	at jenkins.plugins.slack.ActiveNotifier$MessageBuilder.escape(ActiveNotifier.java:526)
	at jenkins.plugins.slack.ActiveNotifier$MessageBuilder.startMessage(ActiveNotifier.java:392)
	at jenkins.plugins.slack.ActiveNotifier$MessageBuilder.<init>(ActiveNotifier.java:307)
	at jenkins.plugins.slack.ActiveNotifier.getBuildStatusMessage(ActiveNotifier.java:269)
	at jenkins.plugins.slack.ActiveNotifier.completed(ActiveNotifier.java:145)
	at jenkins.plugins.slack.SlackNotifier.perform(SlackNotifier.java:452)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:744)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:690)
	at hudson.model.Build$BuildExecution.cleanUp(Build.java:196)
	at hudson.model.Run.execute(Run.java:1857)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:97)
	at hudson.model.Executor.run(Executor.java:429)

**Comment by timja** [comment]
Will need your config from the jobs xml file, look for the slack section and paste it here please

**Comment by timja** [comment]
what's in your global config?

one thing to try would be just click save on the job once, `includeCustomMessage` shouldn't be persisted anymore AFAIK

**Comment by timja** [comment]
Do you have any groovy startup scripts that are modifying the config?
Can you check the config after a save and after a restart?

**Comment by xUrko** [comment]
No, I do not have any groovy scripts. 
The config script stays the same before/after edit and restart. 
I will keep up monitoring for breaking causes....

**Comment by josb** [comment]
```
 <jenkins.plugins.slack.SlackNotifier plugin="slack@2.14">
      <baseUrl></baseUrl>
      <teamDomain></teamDomain>
      <authToken></authToken>
      <tokenCredentialId></tokenCredentialId>
      <botUser>false</botUser>
      <room></room>
      <startNotification>false</startNotification>
      <notifySuccess>true</notifySuccess>
      <notifyAborted>false</notifyAborted>
      <notifyNotBuilt>false</notifyNotBuilt>
      <notifyUnstable>true</notifyUnstable>
      <notifyRegression>false</notifyRegression>
      <notifyFailure>true</notifyFailure>
      <notifyBackToNormal>false</notifyBackToNormal>
      <notifyRepeatedFailure>false</notifyRepeatedFailure>
      <includeTestSummary>false</includeTestSummary>
      <includeFailedTests>false</includeFailedTests>
      <commitInfoChoice>NONE</commitInfoChoice>
      <includeCustomMessage>false</includeCustomMessage>
      <customMessage></customMessage>
      <customMessageSuccess></customMessageSuccess>
      <customMessageAborted></customMessageAborted>
      <customMessageNotBuilt></customMessageNotBuilt>
      <customMessageUnstable></customMessageUnstable>
      <customMessageFailure></customMessageFailure>
    </jenkins.plugins.slack.SlackNotifier>
```

**Comment by timja** [comment]
and what global configuration do you have for slack @josb ?

**Comment by timja** [comment]
can you remove slack compatible app url please? its unneeded and can cause issues

**Comment by timja** [comment]
I suspect it may be this change though:
https://github.com/jenkinsci/slack-plugin/pull/462

**Comment by timja** [comment]
Released as 2.15, may take some time for it to be available in the update centre

**Comment by xUrko** [comment]
I have installed ver. 2.15. I have some bad and good news :). The job is still failing but we are closer to the error source. It looks it's related with Postman's CMD executor, called Newman.  See error....

ERROR: Build step failed with exception
java.lang.NumberFormatException: For input string: "Postman"
	at java.lang.NumberFormatException.forInputString(Unknown Source)
	at java.lang.Integer.parseInt(Unknown Source)
	at java.lang.Integer.parseInt(Unknown Source)
**Caused: java.lang.IllegalArgumentException: can't parse argument number: Postman**
	at java.text.MessageFormat.makeFormat(Unknown Source)
	at java.text.MessageFormat.applyPattern(Unknown Source)
	at java.text.MessageFormat.<init>(Unknown Source)
	at java.text.MessageFormat.format(Unknown Source)
	at jenkins.plugins.slack.ActiveNotifier$MessageBuilder.escape(ActiveNotifier.java:545)
	at jenkins.plugins.slack.ActiveNotifier$MessageBuilder.startMessage(ActiveNotifier.java:392)
	at jenkins.plugins.slack.ActiveNotifier$MessageBuilder.<init>(ActiveNotifier.java:307)
	at jenkins.plugins.slack.ActiveNotifier.getBuildStatusMessage(ActiveNotifier.java:269)
	at jenkins.plugins.slack.ActiveNotifier.completed(ActiveNotifier.java:145)
	at jenkins.plugins.slack.SlackNotifier.perform(SlackNotifier.java:447)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:744)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:690)
	at hudson.model.Build$BuildExecution.cleanUp(Build.java:196)
	at hudson.model.Run.execute(Run.java:1857)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:97)
	at hudson.model.Executor.run(Executor.java:429)
Build step 'Slack Notifications' marked build as failure
